internal/hprof/class.go

```diff
@@ -0,0 +1,636 @@
+}
+
+func readObjectArrayDump(data []byte) (ObjectArrayDump, error) {
+	objectArrayDump := ObjectArrayDump{}
+
+	// Read the array object ID (8 bytes)
+	objectArrayDump.ArrayObjectId = int64(binary.BigEndian.Uint64(data[:8]))
+
+	// Read the stack trace serial number (4 bytes)
+	objectArrayDump.StackTraceSerialNumber = int32(binary.BigEndian.Uint32(data[8:12]))
+
+	// Read the number of elements (4 bytes)
+	objectArrayDump.NumberOfElements = int32(binary.BigEndian.Uint32(data[12:16]))
+
+	// Read the array class object ID (8 bytes)
+	objectArrayDump.ArrayClassObjectId = int64(binary.BigEndian.Uint64(data[16:24]))
+
+	// Read the elements
+	objectArrayDump.Elements = make([]int64, objectArrayDump.NumberOfElements)
+	for i := 0; i < int(objectArrayDump.NumberOfElements); i++ {
+		objectArrayDump.Elements[i] = int64(binary.BigEndian.Uint64(data[24+i*8 : 24+(i+1)*8]))
+	}
+
+	return objectArrayDump, nil
+}
+
+func readPrimitiveArrayDump(data []byte) (PrimitiveArrayDump, error) {
+	primitiveArrayDump := PrimitiveArrayDump{}
+
+	// Read the array object ID (8 bytes)
+	primitiveArrayDump.ArrayObjectId = int64(binary.BigEndian.Uint64(data[:8]))
+
+	// Read the stack trace serial number (4 bytes)
+	primitiveArrayDump.StackTraceSerialNumber = int32(binary.BigEndian.Uint32(data[8:12]))
+
+	// Read the number of elements (4 bytes)
+	primitiveArrayDump.NumberOfElements = int32(binary.BigEndian.Uint32(data[12:16]))
+
+	// Read the element type (1 byte)
+	primitiveArrayDump.ElementType = data[16]
+
+	// Read the elements
+	primitiveArrayDump.Elements = make([]byte, primitiveArrayDump.NumberOfElements)
+	copy(primitiveArrayDump.Elements, data[17:17+primitiveArrayDump.NumberOfElements])
+
+	return primitiveArrayDump, nil
+}
+
+
+func main() {
+	subTagMap := map[byte]func(data []byte) (interface{}, error) {
+		0xFF: func(data []byte) (interface{}, error) { return readRootUnknown(data) },
+		0x01: func(data []byte) (interface{}, error) { return readRootJNIGlobal(data) },
+		0x02: func(data []byte) (interface{}, error) { return readRootJNILocal(data) },
+		0x03: func(data []byte) (interface{}, error) { return readRootJavaFrame(data) },
+		0x04: func(data []byte) (interface{}, error) { return readRootNativeStack(data) },
+		0x05: func(data []byte) (interface{}, error) { return readRootStickyClass(data) },
+		0x06: func(data []byte) (interface{}, error) { return readRootThreadBlock(data) },
+		0x07: func(data []byte) (interface{}, error) { return readRootMonitorUsed(data) },
+		0x08: func(data []byte) (interface{}, error) { return readRootThreadObject(data) },
+		0x20: func(data []byte) (interface{}, error) { return readClassDump(data) },
+		0x21: func(data []byte) (interface{}, error) { return readInstanceDump(data) },
+		0x22: func(data []byte) (interface{}, error) { return readObjectArrayDump(data) },
+		0x23: func(data []byte) (interface{}, error) { return readPrimitiveArrayDump(data) },
+	}
+	subTagMap[0x01](nil) // just to avoid unused error
```

Comment:
1. Why do we need such unused variable in main branch  ?
2. If you need keep unused variable and make compiler happy, idiomatic way is following:
```
_ = sugTagMap
```